service/src/main/java/org/jboss/sbomer/service/rest/criteria/predicate/CustomizedPredicateBuilderStrategy.java

```diff
@@ -108,7 +106,6 @@ public <T> Predicate createPredicate(
     }
 
     private String preprocessLikeOperatorArgument(String argument) {
-        return argument.replaceAll("_", Matcher.quoteReplacement(ESCAPE_CHAR + "_"))
-                .replaceAll("\\" + WILDCARD_CHAR, "%");
+        return argument.replace("_", ESCAPE_CHAR + "_").replaceAll("\\" + WILDCARD_CHAR, "%");
```

Comment:
The first one is not a regex. Does the second one need to be a regex?

Reply:
I need to replace all occurrences of * with %, so yes it could be optimized with `.replace("*", "%")`, but let's note it down and merge this PR.